### PR TITLE
Rel 1.2.0 doc chgs cisco-only

### PR DESCRIPTION
**Pull request by chrisvanheuveln**
netdev changes will be in the next PR

### CODE REVIEW

**Comment by mikewiebe** [comment]
@chrisvanheuveln General comments.  Update the following
- https://github.com/chrisvanheuveln/cisco-network-puppet-module/blob/doc-updates/README.md#artifacts
- We should probably update the version numbers for the gems in the example: https://github.com/chrisvanheuveln/cisco-network-puppet-module/blob/doc-updates/README.md#gems
- Add persistence statement about OAC:  https://github.com/chrisvanheuveln/cisco-network-puppet-module/blob/doc-updates/README.md#gem-persistence-bash-shell-only

File: README.md [hunk 1]
```diff
@@ -142,6 +144,84 @@ cisco_interface_ospf {"Ethernet1/2 Sample":
 }
 ~~~
 
+## Platform Support
+
+### <a name="provider-platform-support">Provider Support Across Platforms</a>
+
+The following table indicates which providers are supported on each platform. As platforms are added to the support list they may indicate ' `Unsupported`' for some providers that have not completed the test validation process at the time of this release. Some providers will show caveats for a platform if there are limitations on usage, such as with unsupported properties or hardware limitations.
```
Comment (mikewiebe) [inline README.md]: IMO, just `Unsupported` would look better without the ''?

File: README.md [hunk 2]
```diff
@@ -142,6 +144,84 @@ cisco_interface_ospf {"Ethernet1/2 Sample":
...
+
+##### Cisco Providers
+
+| ✅ = Supported <br> ❌ = Unsupported | N9k | N30xx | N31xx | N56xx | N6k | N7k | Caveats |
+|:---|:---:|:-----:|:-----:|:-----:|:---:|:---:|:---:|
+| [cisco_aaa_authentication_login](#type-cisco_aaa_authentication_login) | ✅ | ✅ | ✅ | ❌ | ❌ | ❌ |
+| [cisco_aaa_authorization_cfg_svc](#type-cisco_aaa_authorization_cfg_svc) | ✅ | ✅ | ✅ | ❌ | ✅ | ❌ |
```
Comment (mikewiebe) [inline README.md]: While I realize that we do have passing results for beaker on n6k for 1.1.0+ providers, I am not sure I am comfortable with calling them `Supported` without the N(5|7)K results.  Given that N5K runs the same software and N7K also supports OAC, we might get undesired questions about why the other platforms are not also supported?

File: README.md [hunk 3]
```diff
@@ -142,6 +144,84 @@ cisco_interface_ospf {"Ethernet1/2 Sample":
...
+| [cisco_command_config](#type-cisco_command_config) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| [cisco_bgp](#type-cisco_bgp) | ✅ | ✅ | ✅ | ✅* | ✅* | ✅* | * [caveats](#cisco_bgp-caveats) |
+| [cisco_bgp_af](#type-cisco_bgp_af) | ✅* | ✅* | ✅ | ✅ | ✅*  | ✅ | * [caveats](#cisco_bgp_af-caveats) |
+| [cisco_bgp_neighbor](#type-cisco_bgp_neighbor) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| [cisco_bgp_neighbor_af](#type-cisco_bgp_neighbor_af) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| [cisco_interface](#type-cisco_interface) | ✅ | ✅ | ✅ | ✅* | ✅* | ✅ | * [caveats](#cisco_interface-caveats) |
+| [cisco_interface_channel_group](#type-cisco_interface_channel_group) | ✅ | ✅ | ✅ | ❌ | ✅ | ❌ |
+| [cisco_interface_ospf](#type-cisco_interface_ospf) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| [cisco_interface_portchannel](#type-cisco_interface_portchannel) | ✅ | ✅ | ✅ | ❌ | ✅* | ❌ | * [caveats](#cisco_interface_portchannel-caveats) |
+| [cisco_interface_service_vni](#type-cisco_interface_service_vni) | ❌ | ❌ | ❌ | ❌ | ❌ | ✅ |
```
Comment (mikewiebe) [inline README.md]: Did we actually get test results for `cisco_interface_service_vni`? The wiki still says we need to test it.

File: README.md [hunk 4]
```diff
@@ -142,6 +144,84 @@ cisco_interface_ospf {"Ethernet1/2 Sample":
...
+| [cisco_interface_portchannel](#type-cisco_interface_portchannel) | ✅ | ✅ | ✅ | ❌ | ✅* | ❌ | * [caveats](#cisco_interface_portchannel-caveats) |
+| [cisco_interface_service_vni](#type-cisco_interface_service_vni) | ❌ | ❌ | ❌ | ❌ | ❌ | ✅ | 
+| [cisco_ospf](#type-cisco_ospf) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| [cisco_ospf_vrf](#type-cisco_ospf_vrf) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| ✅ = Supported <br> ❌ = Unsupported  | N9k | N30xx | N31xx | N56xx | N6k | N7k | Caveats |
+| [cisco_overlay_global](#type-cisco_overlay_global) | ✅ | ❌ | ❌ | ❌ | ❌ | ❌  |
+| [cisco_pim](#type-cisco_pim) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| [cisco_pim_rp_address](#type-cisco_pim_rp_address) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| [cisco_pim_group_list](#type-cisco_pim_group_list) | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
+| [cisco_portchannel_global](#type-cisco_portchannel_global) | ❌ | ✅ | ✅ | ❌ | ✅ | ❌ |
```
Comment (mikewiebe) [inline README.md]: Need to add a caveat for n5k `cisco_portchannel_global`.  The default value differs for `hash_poly` between n5k and n6k which will cause an impotency issue.

File: README.md [hunk 5]
```diff
@@ -142,6 +144,84 @@ cisco_interface_ospf {"Ethernet1/2 Sample":
...
 
+## Platform Support
+
+### <a name="provider-platform-support">Provider Support Across Platforms</a>
+
+The following table indicates which providers are supported on each platform. As platforms are added to the support list they may indicate ' `Unsupported`' for some providers that have not completed the test validation process at the time of this release. Some providers will show caveats for a platform if there are limitations on usage, such as with unsupported properties or hardware limitations.
+
+##### Cisco Providers
+
+| ✅ = Supported <br> ❌ = Unsupported | N9k | N30xx | N31xx | N56xx | N6k | N7k | Caveats |
```
Comment (mikewiebe) [inline README.md]: For the platforms, we should stick to the `Nxxxx` format instead of `N9k` and `N7k` etc...

File: README.md [hunk 6]
```diff
@@ -323,6 +412,11 @@ The following resources are listed alphabetically.
 
 Allows execution of configuration commands.
 
+| Minimum Requirements | N9k | N30xx | N31xx | N56xx | N6k | N7k |
+|----------------------|:---:|:-----:|:-----:|:-----:|:---:|:---:|
+| OS Image | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.3(0)N1(1) | 7.3(0)N1(1) | 7.3(0)D1(1) |
+| Puppet Module | 1.0.1 | 1.0.1 | 1.0.1 | 1.0.1 | 1.0.1 | 1.0.1 |
```
Comment (mikewiebe) [inline README.md]: 1.2.0 for N(5|6|7) platforms.

File: README.md [hunk 7]
```diff
@@ -346,6 +440,11 @@ This provider allows raw configurations to be managed by Puppet. It serves as a
 
 Manages AAA Authentication Login configuration.
```
Comment (mikewiebe) [inline README.md]: **General Comment** In each minimum requirements table (per provider) we list versions even if they are not currently supported on that platform.  We should add unsupported instead.

File: README.md [hunk 8]
```diff
@@ -1630,11 +1894,111 @@ Whether or not the vlan is shutdown. Valid values are 'true', 'false' and
 keyword 'default'.
 
 --
+### Type: cisco_vpc_domain
```
Comment (mikewiebe) [inline README.md]: Was `cisco_vpc_domain` documentation missing?

File: README.md (hunk 1 above)
Comment (chrisvanheuveln) [inline README.md]: No pref, removed.

File: README.md (hunk 2 above)
Comment (chrisvanheuveln) [inline README.md]: I had the same concerns. I'll X the 6k.

File: README.md (hunk 3 above)
Comment (chrisvanheuveln) [inline README.md]: I tested it just now. Works:

```
        beaker_tests/cisco_interface_service_vni/test_interface_service_vni.rb passed in 109.23 seconds
      Test Suite: tests @ 2016-02-12 17:52:32 -0500

      - Host Configuration Summary -


              - Test Case Summary for suite 'tests' -
       Total Suite Time: 109.23 seconds
      Average Test Time: 109.23 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1
```

File: README.md (hunk 4 above)
Comment (chrisvanheuveln) [inline README.md]: Added.

File: README.md (hunk 6 above)
Comment (chrisvanheuveln) [inline README.md]: Fixed.

File: README.md (hunk 7 above)
Comment (chrisvanheuveln) [inline README.md]: Fixed.

**Pull request merged**